src/react/useQueries.ts

```diff
@@ -41,13 +55,29 @@ export function useQueries(queries: UseQueryOptions[]): UseQueryResult[] {
       mountedRef.current = false
       unsubscribe()
     }
-  }, [observer])
+  }, [someIsLoading])
 
-  React.useEffect(() => {
-    // Do not notify on updates because of changes in the options because
-    // these changes should already be reflected in the optimistic result.
-    observer.setQueries(defaultedQueries, { listeners: false })
-  }, [defaultedQueries, observer])
+  const handleReset = React.useCallback(() => {
+    errorResetBoundary.clearReset()
+    const unsubscribe = observerRef.current.subscribe()
+    throw observerRef.current
+      .refetch({ filter: x => x.getCurrentResult().isLoading })
+      .finally(unsubscribe)
```

Comment:
should we not do something like `fetchOptimistic` does on the QueryObserver, just for multiple queries? We're also not calling `onSuccess / onError / onSettled` like we do in `useBaseQuery` ....